features/metadata/current_scope.feature

```diff
@@ -0,0 +1,84 @@
+Feature: RSpec provides the current_scope as RSpec::current_scope
+
+  You can detect which rspec scope your helper methods or library code is executing in.
+  This is useful if for example, your method only makes sense to call in a certain context.
+
+  Scenario: Detecting the current scope
+    Given a file named "current_scope_spec.rb" with:
+      """ruby
+      unless RSpec.current_scope == :suite
+        raise "bad current scope: #{RSpec.current_scope.inspect}"
+      end
```

Comment:
I don't think this is necessary

```suggestion
```

Reply:
That seems to be the only way to test that it's `:suite` at this stage. It's kind of a weird "test" though.

Reply:
It gave me reasonable feedback that things were not working correctly while working on it.

I'm not sure how else to test it.

Similarly I don't know how you would test what value it has after the suite is finished.

Reply:
> what value it has after the suite is finished

Will `at_exit { exit(1) unless RSpec.current_scope.nil? }` work?

Reply:
> Will at_exit { exit(1) unless RSpec.current_scope.nil? } work?

Nice. Didn't know I could do that.